pkg/api-server/auth_test.go

```diff
@@ -25,48 +24,39 @@ var _ = Describe("Auth test", func() {
 	var httpsClientWithoutCerts *http.Client
 	var httpPort uint32
 	var httpsPort uint32
-	var stop chan struct{}
+	var stop = func() {}
 	var externalIP string
 
 	BeforeEach(func() {
 		externalIP = getExternalIP()
 		Expect(externalIP).ToNot(BeEmpty())
 		certPath, keyPath := createCertsForIP(externalIP)
-
-		resourceStore := memory.NewStore()
-		metrics, err := metrics.NewMetrics("Standalone")
-		Expect(err).ToNot(HaveOccurred())
-
-		cfg := config.DefaultApiServerConfig()
-		cfg.HTTPS.TlsCertFile = certPath
-		cfg.HTTPS.TlsKeyFile = keyPath
-		cfg.Authn.Type = certs.PluginName
-		cfg.Auth.ClientCertsDir = filepath.Join("..", "..", "test", "certs", "client")
-		apiServer := createTestApiServer(resourceStore, cfg, true, metrics)
+		var apiServer *api_server.ApiServer
+		Eventually(func() (err error) {
+			apiServer, stop, err = TryStartApiServer(NewTestApiServerConfigurer().WithConfigMutator(func(cfg *config.ApiServerConfig) {
+				cfg.HTTPS.TlsCertFile = certPath
+				cfg.HTTPS.TlsKeyFile = keyPath
+				cfg.Authn.Type = certs.PluginName
+				cfg.Auth.ClientCertsDir = filepath.Join("..", "..", "test", "certs", "client")
+			}))
+			return
+		}).Should(Succeed())
```

Comment:
default timeout is 1s and interval is 10ms. Is that enough? I think we should make it more like 10s, 100ms